File: uber/src/main/java/com/minageorge/uber/store/model/markeritem/MarkerDao.java
package com.minageorge.uber.store.model.markeritem;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import io.reactivex.Flowable;

@Dao
public interface MarkerDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void upsert(List<MarkerEntity> markers);

    @Query("select * from marker")
    Flowable<List<MarkerEntity>> getMarkers();

}

File: uber/src/main/java/com/minageorge/uber/store/model/markermodel/MarkerDao.kt
package com.minageorge.uber.store.model.markermodel

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Flowable

@Dao
interface MarkerDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun upsert(markers: List<MarkerEntity>)

    @Query("select * from markers")
    fun getMarkers(): Flowable<List<MarkerEntity>>
}